Server/JokesOnYou.Web.Api/Repositories/Interfaces/IJokesRepository.cs

```diff
@@ -9,7 +9,10 @@ namespace JokesOnYou.Web.Api.Repositories.Interfaces
 {
     public interface IJokesRepository
     {
-        public Task<IEnumerable<Joke>> GetAllJokesAsync();
-        public Task<IEnumerable<JokeReplyDto>> GetAllJokeDtosAsync();
+        public Task CreateJokeAsync(Joke joke);
+        public Task<List<Joke>> GetAllJokesAsync();
+        public Task<bool> DoesJokeExist(JokeCreateDto jokeCreateDto);
```

Comment:
cant we check if a joke exists without having to pass down a jokecreatedto?

Reply:
how ? you have to know what to look for to be able to know if it exists?

Reply:
Looked at your method so all you would need is a normalized premise and punchline to be sent down, making this method reusable not limited to a specific dto

Reply:
duh :P